config/default.json

```diff
@@ -0,0 +1,78 @@
+{
+    "base" : {
+        "writable_misa" : true,
+        "writable_fiom" : true,
+        "writable_hpm_counters" : {
+            "len" : 32,
+            "value" : "0xFFFF_FFFF"
+        },
+        "mtval_has_illegal_instruction_bits" : false
+    },
+    "memory" : {
+        "pmp" : {
+            "grain" : 0,
+            "count" : 0
+        },
+        "misaligned" : {
+            "enabled" : false
+        },
+        "translation" : {
+            "dirty_update" : false
+        }
+    },
+    "platform" : {
+        "reset_vector" : 4096,
+        "cache_block_size_exp" : 6,
+        "ram" : {
+            "base" : 2147483648,
+            "size" : 2147483648
+        },
+        "rom" : {
+            "base" : 4096,
+            "size" : 4096
+        },
+        "clint" : {
+            "base" : 33554432,
+            "size" : 786432
```

Comment:
Does the JSON parser support hex literals? IMO this would be easier to read.

Reply:
It's using cJSON, which looks like it does (uses strtod), but then Sail itself re-converts the string via mpz_set_str and hard-codes a base of 10: https://github.com/rems-project/sail/blob/ca6092efe10b5c915185cfc82db9cf560901408d/lib/sail_config.c#L260

Also, JSON doesn't officially have hex literals anyway, so that would be relying on an extension.

But then using JSON numbers for these things might be a bad idea anyway, although JSON itself places no restrictions on the ranges of numbers, implementations often behave like JavaScript and treat everything as a double, so can't represent all possible 64-bit integers. cJSON does, but Sail is extracting the string and re-parsing it, so I guess it's not subject to that limitation, though it's a somewhat gross way for it to work.

Reply:
If it's a Sail bitvector you can encode it as a string, but Sail integers are always JSON integers. The problem with using numbers encoded as strings is you can't really say much (really anything) useful about them in a Schema.

The JSON should be standard JSON without any extensions so Schema validators can read it (Arbitrary precision support is mandatory in JSON schema implementations, so issues with silently truncating to doubles won't happen there)